### PR TITLE
W21 103 filesystem renaming

### PR DESCRIPTION
**Pull request by omeh-a**
This PR adds a route for renaming items in the filesystem. Also made several small changes in the backend to neaten things up - e.g. getting rid of redundant breaks in switches and formatting. I wrote some prototype code for getting children but this is left commented out.

### CODE REVIEW

**Comment by Varun-Sethu** [comment]
is this constraint actually enforced anywhere?

File: Backend/filesystem/endpoints.go [hunk 1]
```diff
@@ -116,3 +114,43 @@ func DeleteFilesystemEntity(w http.ResponseWriter, r *http.Request) {
...
+// Handler for renaming filesystem entities
+func RenameFilesystemEntity(w http.ResponseWriter, r *http.Request) {
+	var input ValidRenameRequest
+	if validRequest := httpUtil.ParseParamsToSchema(w, r, []string{"POST"}, map[int]string{
+		400: "missing paramaters, must have: NewName, EntityID",
+		405: "invalid method",
+	}, &input); validRequest {
+		err := renameEntity(httpDbPool, input.EntityID, input.NewName)
+		if err != nil {
+			httpUtil.ThrowRequestError(w, 500, "unable rename, the requested name is taken or this item is the root directory")
```
Comment (Varun-Sethu) [inline Backend/filesystem/endpoints.go:151]: is this constraint actually enforced anywhere?

File: Backend/filesystem/service.go [hunk 2]
```diff
@@ -92,3 +92,31 @@ func deleteEntity(pool database.Pool, entityID int) error {
 	}
 	return nil
 }
+
+// renameEntity changes the logical name of a given object in the filesystem
+func renameEntity(pool database.Pool, entityID int, newName string) error {
+	_, err := pool.GetConn().Exec(context.Background(),
+		"UPDATE filesystem SET logicalname = ($1) WHERE entityid = ($2)", newName, entityID)
+	if err != nil {
```
Comment (Varun-Sethu) [inline Backend/filesystem/service.go:100]: can just:
```go
return err
```

File: Backend/filesystem/service.go [hunk 3]
```diff
@@ -92,3 +92,31 @@ func deleteEntity(pool database.Pool, entityID int) error {
...
+	}
+	return nil
+}
+
+// // getFilesystemInfo returns information regarding a specific file system entity
+// func getEntityChildren(pool database.Pool, docID int) (EntityInfo, error) {
+// 	entity := EntityInfo{}
+// 	children := pgtype.Hstore{}
+
+// 	err := pool.GetConn().QueryRow(context.Background(), "SELECT EntityID FROM filesystem WHERE Parent = $1",
```
Comment (Varun-Sethu) [inline Backend/filesystem/service.go:111]: this isnt what `getEntityChildren` is meant to do, I explained it to nian on discord, it should be there

File: Backend/filesystem/service_test.go [hunk 4]
```diff
@@ -122,3 +122,21 @@ func TestEntityDeletion(t *testing.T) {
 	root, _ = getRootInfo(pool)
 	assert.NotContains(strconv.Itoa(anotherDirectory), root.Children)
 }
+
+func TestEntityRename(t *testing.T) {
```
Comment (Varun-Sethu) [inline Backend/filesystem/service_test.go:126]: maybe add a test case for an entity ID that doesnt exist?

**Review by Varun-Sethu — APPROVED** [review]
just some minor changes, make sure you find a way to enforce that root cannot have its name changed before u merge

File: Backend/filesystem/endpoints.go (hunk 1 above)
Comment (Varun-Sethu) [reply Backend/filesystem/endpoints.go:151]: im referring to: "item is the root directory"

File: Backend/filesystem/service.go (hunk 2 above)
Comment (Varun-Sethu) [reply Backend/filesystem/service.go:100]: can you also update: https://github.com/csesoc/cms.csesoc.unsw.edu.au/blob/prototype/Backend/filesystem/service.go#L87 when you do that? I realised i did the same thing lmao

Comment (omeh-a) [reply Backend/filesystem/service.go:100]: yup - only did this because you did it too

File: Backend/filesystem/service.go (hunk 3 above)
Comment (omeh-a) [reply Backend/filesystem/service.go:111]: this is just the getInfo function copy pastad - i stopped when i asked you about the db, will be different for actual implementation

Comment (Varun-Sethu) [reply Backend/filesystem/service.go:105]: aight cool

**Review by omeh-a — COMMENTED** [review]
> 
> 
> just some minor changes, make sure you find a way to enforce that root cannot have its name changed before u merge

Given that the name of root doesn't matter really i just changed this message. There are no DB constraints against renaming it anyway.

**Pull request merged**